types.ts

```diff
@@ -413,6 +418,76 @@ export interface WebBasicAuthentication extends BaseAuthentication {
   };
 }
 
+/**
+ * Parameters for the {@link CustomAuthentication} authentication type.
+ */
+export interface CustomAuthParameter {
+  /**
+   * The name used to refer to this parameter and to generate the template replacement string.
+   */
+  name: string;
+
+  /**
+   * A description shown to the user indicating what value they should provide for this parameter.
+   */
+  description: string;
+}
+
+/**
+ * Authenticate for custom, non-standard API authentication schemes by inserting one or more arbitrary secret values
+ * into the request (the body, URL, headers, or form data) using template replacement.
+ *
+ * Some APIs use non-standard authentication schemes which often require secret credentials to be put in specific places
+ * in the request URL or request body. Custom authentication supports many of these cases by allowing you as the pack
+ * author to define one or more secret values that the user or you as the pack author must provide (depending on
+ * user or system authentication). When constructing a network request, you may indicate where these values should
+ * be inserted by our fetcher service using the syntax described below (similar to templating engines).
+ *
+ * {% raw %}
```

Comment:
this is needed because otherwise `mkdocs` tries to templatize the `{{...}}` and errors out in rendering the markdown.